ddp-workspace/projects/ddp-atcp/src/app/components/activityForm/app-atcp-activity.component.ts

```diff
@@ -0,0 +1,315 @@
+                                    (click)="close()">
+                                    <mat-icon class="button__icon">highlight_off</mat-icon>
+                                    {{'SDK.NotAgreeButton' | translate}}
+                            </button>
+                            <button #submitButton
+                                    [disabled]="(isPageBusy | async) || dataEntryDisabled"
+                                    class="button button_medium button_primary"
+                                    (click)="flush()"
+                                    (mouseenter)="mouseEnterOnSubmit()">
+                                    <mat-icon *ngIf="!(isPageBusy | async)" class="button__icon">check_circle_outline</mat-icon>
+                                    {{(isPageBusy | async)
+                                        ? ('SDK.SavingButton' | translate)
+                                        : ('SDK.AgreeButton' | translate)}}
+                            </button>
+                        </ng-container>
+                        <ng-container *ngIf="!isStepped || isLastStep && (model.readonly && isLoaded)">
+                            <button *ngIf="model.readonly && isLoaded"
+                                    [ngClass]="{'ButtonBordered  ButtonBordered--green button_right' : model.activityCode !== ActivityCodes.ASSENT,
+                                         'ButtonFilled ButtonFilled--kids ButtonFilled--light-green' : model.activityCode === ActivityCodes.ASSENT}"
+                                    (click)="navigateToConsole()"
+                                    [innerHTML]="'SDK.CloseButton' | translate">
+                            </button>
+                        </ng-container>
+                    </div>
+                    <div *ngIf="displayGlobalError$ | async" class="ErrorMessage">
+                        <span translate>SDK.ValidateError</span>
+                    </div>
+                    <div *ngIf="communicationErrorOccurred" class="ErrorMessage">
+                        <span translate>SDK.CommunicationError</span>
+                    </div>
+                </ng-container>
+            </div>
+        </section>
+      </main>`,
+  providers: [SubmitAnnouncementService, SubmissionManager]
+})
+export class AtcpActivityComponent extends AtcpActivityBaseComponent implements OnInit, OnDestroy, AfterViewInit {
+  @Input() agreeConsent = false;
+  @Input() buttonWithArrow = false;
+
+  public ActivityCodes = ActivityCodes;
+  constructor(
+      windowRef: WindowRef,
+      renderer: Renderer2,
+      submitService: SubmitAnnouncementService,
+      analytics: AnalyticsEventsService,
+      @Inject(DOCUMENT) document: any,
+      injector: Injector) {
+      super(windowRef, renderer, submitService, analytics, document, injector);
+  }
+
+  public isAgree(): boolean {
+      return this.model.formType === 'CONSENT' && this.agreeConsent;
+  }
```

Comment:
Do you really need this method? We needed to add Agree buttons for the TestBoston study.

Reply:
If we are going to move our version of activity into SDK, then it is better to keep this file as much similar as activity in SDK.